lib/faraday/adapter_registry.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'monitor'
+
+module Faraday
+  # AdapterRegistry registers adapter class names so they can be looked up by a
+  # String or Symbol name.
+  class AdapterRegistry
+    def initialize
+      @lock = Monitor.new
+      @constants = nil
```

Comment:
If we initialize `@constants` to `Hash.new` here, a few other lines can disappear.

Reply:
Good catch, initializing it to `{}` or `Hash.new` here, we can avoid testing its initialization on `get` and `set`